src/entity/presentation/presentationWrapperEntity.ts

```diff
@@ -1,20 +1,54 @@
+import { IsDefined, IsNotEmpty } from 'class-validator';
 import { Column, Entity, ObjectID, ObjectIdColumn } from 'typeorm';
 
 import { CallbackEntity } from '../callbackEntity';
 import { ChallengeEntity } from '../challengeEntity';
+import { ThreadEntity } from '../threadEntity';
 
 @Entity('presentation_wrapper')
 export class PresentationWrapperEntity {
   @ObjectIdColumn()
   // @ts-ignore
   _id: ObjectID;
+
   @Column()
+  @IsNotEmpty({ message: 'PresentationWrapper.pdId is invalid' })
   // @ts-ignore
-  callback: CallbackEntity;
+  pdId: string;
+
   @Column()
+  @IsDefined({ message: 'PresentationWrapper.challenge must be provided' })
   // @ts-ignore
   challenge: ChallengeEntity;
+
   @Column()
+  @IsDefined({ message: 'PresentationWrapper.presentation must be provided' })
   // @ts-ignore
-  presentation: PresentationEntity
+  presentation: PresentationEntity;
+
+  @Column()
+  @IsDefined({ message: 'PresentationWrapper.thread must be provided' })
+  // @ts-ignore
+  thread: ThreadEntity;
+
+  @Column()
+  @IsNotEmpty({ message: 'PresentationDefinitionWrapper.id is invalid' })
+  // @ts-ignore
+  id: string;
```

Comment:
PK

Reply:
That one needs to match the presentation id afaik, so I will leave it as it is for the same reasons as above.